src/extractor/ConfigurationExtractor.ts

```diff
@@ -36,9 +36,27 @@ export class ConfigurationExtractor {
       (ConfigurationExtractor.inferCanonical(resources) || 'http://example.org');
     if (!fhirVersion?.length) {
       const fhirVersionFromResources = ConfigurationExtractor.inferString(resources, 'fhirVersion');
-      fhirVersion = utils.isSupportedFHIRVersion(fhirVersionFromResources)
-        ? [fhirVersionFromResources]
-        : ['4.0.1'];
+      if (utils.isSupportedFHIRVersion(fhirVersionFromResources)) {
+        fhirVersion = [fhirVersionFromResources];
+      }
+    }
+    if (specifiedFHIRVersion) {
+      // If there is a detected version, check if it matches the specified version
+      if (fhirVersion?.length && fhirVersion.indexOf(specifiedFHIRVersion) === -1) {
+        logger.warn(
+          `FHIR Version mismatch error: specified version is: ${specifiedFHIRVersion} while detected version is: ${fhirVersion.join(
+            ', '
+          )}`
+        );
+        logger.warn(
+          `GoFSH will use the specified version (${specifiedFHIRVersion}), but this may result in additional processing errors.`
+        );
```

Comment:
Instead of logging this as a _second_ warning, I'd just add this sentence to the end of the warning above -- because it's really the same basic thing.  That way it doesn't get logged w/ a warning count of 2.